Remove VP9 preference code from PlatformMediaSessionManager

eric-carlson opened the pull request:
#### 9c90034fa4439d65df4f3b2f5001bcc79ee73fec
<pre>
Remove VP9 preference code from PlatformMediaSessionManager
<a href="https://bugs.webkit.org/show_bug.cgi?id=291034">https://bugs.webkit.org/show_bug.cgi?id=291034</a>
<a href="https://rdar.apple.com/148553383">rdar://148553383</a>

Reviewed by Jer Noble and Jean Yves Avenard.

The VP9 preference code in PlatformMediaSessionManager was partially duplicated in
VP9Utilities, duplicate the rest and remove it all from PlatformMediaSessionManager.

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::setShouldEnableVP9Decoder): Deleted.
(WebCore::PlatformMediaSessionManager::shouldEnableVP9Decoder): Deleted.
(WebCore::PlatformMediaSessionManager::setSWVPDecodersAlwaysEnabled): Deleted.
(WebCore::PlatformMediaSessionManager::swVPDecodersAlwaysEnabled): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::ensureCodecsRegistered):
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.h:
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.mm:
(WebCore::VP9TestingOverrides::setShouldEnableVP9Decoder):
(WebCore::VP9TestingOverrides::shouldEnableVP9Decoder const):
(WebCore::shouldEnableVP9Decoder):
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::updateGPUProcessPreferences):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_textAnimationController):
(WebKit::WebPage::updatePreferences):
</pre>

jyavenard commented on Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
```diff
@@ -78,9 +78,9 @@
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        if (shouldEnableVP9Decoder())
+        if (WebCore::shouldEnableVP9Decoder())
```
Probably don't need `WebCore` here

jyavenard commented on Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
```diff
@@ -78,9 +78,9 @@
+        if (WebCore::shouldEnableVP9Decoder())
             registerSupplementalVP9Decoder();
-        if (shouldEnableSWVP9Decoder())
+        if (WebCore::shouldEnableSWVP9Decoder())
```
ditto

jyavenard reviewed (approved):
r+ ; however I believe those prefs are non functional to start. It only make MediaCapabilities pretend that we do support VP9 when we do not.
They don't make vp9 works under any circumstances when running on a platform that didn't have vp9 to start (such as iOS simulator or Intel Mac Pro)

I would remove those prefs entirely and mark the tests as failing because what they are testing is nonsensical.

eric-carlson committed:
Remove VP9 preference code from PlatformMediaSessionManager
https://bugs.webkit.org/show_bug.cgi?id=291034
rdar://148553383

Reviewed by Jean-Yves Avenard and Jer Noble.

The VP9 preference code in PlatformMediaSessionManager was partially duplicated in
VP9Utilities, duplicate the rest and remove it all from PlatformMediaSessionManager.

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::setShouldEnableVP9Decoder): Deleted.
(WebCore::PlatformMediaSessionManager::shouldEnableVP9Decoder): Deleted.
(WebCore::PlatformMediaSessionManager::setSWVPDecodersAlwaysEnabled): Deleted.
(WebCore::PlatformMediaSessionManager::swVPDecodersAlwaysEnabled): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::ensureCodecsRegistered):
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.h:
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.mm:
(WebCore::VP9TestingOverrides::setShouldEnableVP9Decoder):
(WebCore::VP9TestingOverrides::shouldEnableVP9Decoder const):
(WebCore::shouldEnableVP9Decoder):
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::updateGPUProcessPreferences):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_textAnimationController):
(WebKit::WebPage::updatePreferences):

Canonical link: https://commits.webkit.org/293247@main
